hyp3_sdk/util.py

```diff
@@ -1,17 +1,70 @@
 """Extra utilities for working with HyP3"""
+from pathlib import Path
 
 import requests
+from requests.adapters import HTTPAdapter
+from urllib3.util.retry import Retry
+
+import hyp3_sdk
+from hyp3_sdk.exceptions import AuthenticationError
 
 AUTH_URL = 'https://urs.earthdata.nasa.gov/oauth/authorize?response_type=code&client_id=BO_n7nTIlMljdvU6kRRB3g' \
            '&redirect_uri=https://auth.asf.alaska.edu/login'
 
 
-def get_authenticated_session() -> requests.Session:
-    """logs into hyp3 using credentials for urs.earthdata.nasa.gov from a .netrc file.
+def get_authenticated_session(username: str, password: str) -> requests.Session:
+    """logs into hyp3 using credentials for urs.earthdata.nasa.gov from provided credentails or a .netrc file.
 
     Returns:
-        An authenticated session
+        An authenticated Session object from the requests library
     """
     s = requests.Session()
```

Comment:
I'd love to see `s.headers.update('User-Agent', f'hyp3-sdk {hyp3_sdk.__version__}')` here, that should let us review API logs to determine if anyone is actually using the SDK.  Otherwise the user agent will be a generic `requests 2.25.0` or the like.